File: new_server/vueblog-common/src/main/java/cn/smilex/vueblog/common/annotation/CrossOrigin.java
package cn.smilex.vueblog.common.annotation;

import com.linecorp.armeria.server.annotation.AdditionalHeader;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author smilex
 * @date 2022/11/12/17:20
 * @since 1.0
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
@AdditionalHeader(name = "Access-Control-Allow-Origin", value = "*")
@AdditionalHeader(name = "Access-Control-Allow-Credentials", value = "true")
@AdditionalHeader(name = "Access-Control-Allow-Methods", value = "*")
@AdditionalHeader(name = "Access-Control-Allow-Headers", value = "*")
@AdditionalHeader(name = "Access-Control-Expose-Headers", value = "*")
public @interface CrossOrigin {
}

File: new_server/vueblog-common/src/main/java/vip/smilex/vueblog/common/annotation/CrossOrigin.kt
package vip.smilex.vueblog.common.annotation

import com.linecorp.armeria.server.annotation.AdditionalHeader

/**
 * @author smilex
 * @date 2023/12/4 21:45:31
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.CLASS)
@AdditionalHeader(name = "Access-Control-Allow-Origin", value = ["*"])
@AdditionalHeader(name = "Access-Control-Allow-Credentials", value = ["true"])
@AdditionalHeader(name = "Access-Control-Allow-Methods", value = ["*"])
@AdditionalHeader(name = "Access-Control-Allow-Headers", value = ["*"])
@AdditionalHeader(name = "Access-Control-Expose-Headers", value = ["*"])
annotation class CrossOrigin()
